### 1. 
Les KPI sont des indicateurs qui permettent de mesurer une performance

exemple:
    Nombre de personnes ayant visité votre site en comparant avec les années/mois précédent. Cela permettra d'analyser l'evolution du trafic d'un site.
    

In [28]:
import pandas as pd
import numpy as np

In [29]:
df = pd.read_csv('../../data/eval_data.csv') #Permet de lire un fichier .csv

In [30]:
df.head(10)

Unnamed: 0  order_id  quantity                              item_name  \
0           0         1         1           Chips and Fresh Tomato Salsa   
1           1         1         1                                   Izze   
2           2         1         1                       Nantucket Nectar   
3           3         1         1  Chips and Tomatillo-Green Chili Salsa   
4           4         2         2                           Chicken Bowl   
5           5         3         1                           Chicken Bowl   
6           6         3         1                          Side of Chips   
7           7         4         1                          Steak Burrito   
8           8         4         1                       Steak Soft Tacos   
9           9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN        NaN  
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN        NaN  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

In [31]:
df.ndim #Afficher la dimension du dataset

2

In [39]:
df.info() #Les colonnes correspondent à celle d'une base de données d'une entreprise de restauration. Certaine colonne sont de types object. Il faudra probablement adapter leur type si l'on souhaite les utilisées.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Unnamed: 0          4622 non-null   int64 
 1   order_id            4622 non-null   int64 
 2   quantity            4622 non-null   int64 
 3   item_name           4622 non-null   object
 4   choice_description  3376 non-null   object
 5   item_price          4522 non-null   object
dtypes: int64(3), object(3)
memory usage: 216.8+ KB


In [33]:
df

Unnamed: 0  order_id  quantity                              item_name  \
0              0         1         1           Chips and Fresh Tomato Salsa   
1              1         1         1                                   Izze   
2              2         1         1                       Nantucket Nectar   
3              3         1         1  Chips and Tomatillo-Green Chili Salsa   
4              4         2         2                           Chicken Bowl   
...          ...       ...       ...                                    ...   
4617        4617      1833         1                          Steak Burrito   
4618        4618      1833         1                          Steak Burrito   
4619        4619      1834         1                     Chicken Salad Bowl   
4620        4620      1834         1                     Chicken Salad Bowl   
4621        4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN        NaN  
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN        NaN  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 6 columns]

In [34]:
df.describe()

Unnamed: 0     order_id     quantity
count  4622.000000  4622.000000  4622.000000
mean   2310.500000   927.254868     1.075725
std    1334.400802   528.890796     0.410186
min       0.000000     1.000000     1.000000
25%    1155.250000   477.250000     1.000000
50%    2310.500000   926.000000     1.000000
75%    3465.750000  1393.000000     1.000000
max    4621.000000  1834.000000    15.000000

In [35]:
print(df.loc[[4621]])

      Unnamed: 0  order_id  quantity           item_name  \
4621        4621      1834         1  Chicken Salad Bowl   

                                     choice_description item_price  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   


D'après les statistiques descriptives du dataset, l'item le plus commandé est l'item 4621 soit:
    Le Chicken Salad Bowl (avec 1834 commande)
car c'est celui-ci qui possède la relation commande.

D'après notre dataset, on peut voir qu'il y a au total 4621 identifiants soit 4621 commande.

In [36]:
df.sum(axis = 0, skipna = True) 

Unnamed: 0                                             10679131
order_id                                                4285772
quantity                                                   4972
item_name     Chips and Fresh Tomato SalsaIzzeNantucket Nect...
dtype: object

In [37]:
df['item_price']=df['item_price'].str[:1]

In [38]:
df

Unnamed: 0  order_id  quantity                              item_name  \
0              0         1         1           Chips and Fresh Tomato Salsa   
1              1         1         1                                   Izze   
2              2         1         1                       Nantucket Nectar   
3              3         1         1  Chips and Tomatillo-Green Chili Salsa   
4              4         2         2                           Chicken Bowl   
...          ...       ...       ...                                    ...   
4617        4617      1833         1                          Steak Burrito   
4618        4618      1833         1                          Steak Burrito   
4619        4619      1834         1                     Chicken Salad Bowl   
4620        4620      1834         1                     Chicken Salad Bowl   
4621        4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN        NaN  
1                                          [Clementine]          $  
2                                               [Apple]          $  
3                                                   NaN        NaN  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...          $  
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...          $  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...          $  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...          $  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...          $  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...          $  

[4622 rows x 6 columns]